src/main/java/io/gravitee/resource/oauth2/generic/OAuth2GenericResource.java

```diff
@@ -351,7 +349,16 @@ public void handle(AsyncResult<HttpClientResponse> asyncResponse) {
                                                 if (response.statusCode() == HttpStatusCode.OK_200) {
                                                     responseHandler.handle(new UserInfoResponse(true, buffer.toString()));
                                                 } else {
-                                                    responseHandler.handle(new UserInfoResponse(false, buffer.toString()));
+                                                    logger.error(
+                                                        "An error occurs while getting userinfo from access token. Request ends with status {}: {}",
+                                                        response.statusCode(),
+                                                        buffer.toString()
+                                                    );
+                                                    responseHandler.handle(
+                                                        new UserInfoResponse(
+                                                            new Throwable("An error occurs while getting userinfo from access token")
```

Comment:
Same here > That the limit I pointed out yesterday, I would prefer having an explicit exception here than a generic Throwable.

Reply:
Use now `OAuth2ResourceException`